scripts/invitejoin.pl

```diff
@@ -33,41 +33,213 @@
+                push(@allowed_nicks, $new_allowed);
+            }
+        }
+        close(F);
+    }
+}
+
+sub save_allowed_nicks {
+    my ($file) = @_;
+
+    if (-e $file) {
+        local *F;
+        open(F, ">", $file);
+
+        for my $allowed (@allowed_nicks) {
+            print(F join("\t", $allowed->{net}, $allowed->{nick}) . "\n");
+        }
+    
+        close(F);
+    } else {
+      open(F, ">", $file) or die "Can't create $file. Reason: $!";
+      close(F);
+      save_allowed_nicks($file);
+    }
+}
+
+sub new_allowed_nick {
+    my $anick = {};
+
+    $anick->{net}  = shift;
+    $anick->{nick} = shift;
+
+    return $anick;
+}
+
+sub add_allowed_nick {
+    my ($network, $nick) = split(" ", $_[0], 2);
+    my ($correct_net);
+    
+    if ($network eq "" || $nick eq "") {
+        Irssi::printformat(MSGLEVEL_CLIENTCRAP, 'invitejoin_usage_add_nick');
+        return;
+    }
     
+    if ($network) {
+        my ($ircnet) = Irssi::chatnet_find($network);
+        if (!$ircnet) {
+            Irssi::printformat(MSGLEVEL_CLIENTCRAP, 'invitejoin_no_net', $network);
+            return;
+        } else {
+            $correct_net = 1;
+        }
+    }
+ 
+    if ($correct_net && $nick) {
+        push(@allowed_nicks, new_allowed_nick($network, $nick));
+        save_allowed_nicks("$irssidir/$allowed_nicks_file");
+            
+        Irssi::printformat(MSGLEVEL_CLIENTCRAP, 'saved_nick', $network, $nick);
+    }
+}
+
+sub del_allowed_nick {
+    my ($ircnet, $nick) = split(" ", $_[0], 2);
+
+    if ($ircnet eq "") {
+        Irssi::printformat(MSGLEVEL_CLIENTCRAP, 'invitejoin_delusage');
+        return;
+    }
+
+    my $size_before = scalar(@allowed_nicks);
+    @allowed_nicks = grep { ! ($_->{net} eq $ircnet && $_->{nick} eq $nick) } @allowed_nicks;
+    my $size_after = scalar(@allowed_nicks);
+    
+    if ($size_after != $size_before) {
+        Irssi::printformat(MSGLEVEL_CLIENTCRAP, 'invitejoin_delled', $ircnet, $nick);
+        save_allowed_nicks("$irssidir/$allowed_nicks_file");
+    } else {
+        Irssi::printformat(MSGLEVEL_CLIENTCRAP, 'invitejoin_nfound', $ircnet, $nick);
+    }
+}
+
+sub list_allowed_nicks {
+    
+    if (@allowed_nicks == 0) {
+        Irssi::printformat(MSGLEVEL_CLIENTCRAP, 'allowed_nicks_empty');
+    } else {
+        Irssi::printformat(MSGLEVEL_CLIENTCRAP, 'allowed_nicks_info');
+
+        for my $allowed (@allowed_nicks) {
+            Irssi::printformat(MSGLEVEL_CLIENTCRAP, 'allowed_nicks_print', $allowed->{net}, $allowed->{nick});
+        }
+    }
+}
+
+sub invitejoin_runsub {
+    
+    my ($data, $server, $item) = @_;
+    $data =~ s/\s+$//g;
+    
+    if ($data) {
+        Irssi::command_runsub('invitejoin', $data, $server, $item);
+    } else {
+        Irssi::printformat(MSGLEVEL_CLIENTCRAP, 'invitejoin_usage');
+    }
+}
+
+sub is_allowed_nick {
+  my ($server, $nick) = @_;
+
+  return 1 unless @allowed_nicks < 1;
```

Comment:
Not quite.

It keeps the previous behavior which allows all invites without restriction. If you have set any allowed nicks, it restricts the invite joins to the allowed nicks.

> so this will always return true if there are**n't** any allowed nicks, but always fail if there are no **matching** allowed nicks?